lib/jsdom/living/nodes/HTMLScriptElement-impl.js

```diff
@@ -47,15 +54,13 @@ class HTMLScriptElementImpl extends HTMLElementImpl {
     }
   }
 
-  // Also used during parsing as a hack around proper script evaluation.
-  //
-  // Note: because of how hacky this all is, this is generally called twice during initial parsing cases: 1) When the
-  // element is inserted into the document (when its start tag is encountered), when there's no text content and no
-  // src="" attribute. 2a) Once the src="" attribute is encountered, if applicable. 2b) Once the end tag is encountered
-  // and there is interesting child text content, if applicable. Note that both 2a) and 2b) can be true in strange
-  // cases (see e.g. the html/semantics/scripting-1/the-script-element/execution-timing/115.html WPT).
+  _poppedOffStackOfOpenElements() {
+    this._closeTagHasBeenSeen = true;
+    this._eval();
+  }
+
   _eval() {
-    if (!this._attached || this._startedEval) {
+    if ((!this._closeTagHasBeenSeen && this._parserInserted) || !this._attached || this._startedEval) {
```

Comment:
Same as `<style>`: I'd rather the `!this._parserInserted` be placed in `_attach()`. This should also avoid needing to add the `_closeTagHasBeenSeen` flag.

See https://html.spec.whatwg.org/multipage/scripting.html#script-processing-model:parser-inserted-2, which has:

> When a `script` element that is **not marked as being "parser-inserted"** experiences one of the events listed in the following list, the user agent must immediately prepare the script element:
>
> - The `script` element **becomes connected**.
> - ...

Reply:
I was trying to do a more direct translation of pre-8c84ebf3b7f3420b6e6cc68612ed3e0b9c1f4455, but hopefully it works to follow the spec too. Let's give it a shot...